src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskAsyncEnumerableExtensions.ToBlockingEnumerable.cs

```diff
@@ -25,6 +25,9 @@ public static partial class TaskAsyncEnumerableExtensions
         /// </remarks>
         [UnsupportedOSPlatform("browser")]
         public static IEnumerable<T> ToBlockingEnumerable<T>(this IAsyncEnumerable<T> source, CancellationToken cancellationToken = default)
```

Comment:
This method is troublesome. Its implementation uses `yield return` which can't be performed for `ref struct`. It should be excluded.

Reply:
`yield` could be avoided by implementing a custom `IEnumerable<T>` implementation (which is trivial), but if you care about performance so much to use ref structs, you should avoid blocking threads too. 😅

Reply:
yeah I keep forgetting C# yield will store T as member no matter what and compile fine even with T allows ref struct. I have removed it again.

`ref struct` isn't necessarily about performance, nor is async/away (scale), but really about correctness in the face of reusing/returning internal resources.